src/DataObjects/Document.php

```diff
@@ -375,10 +375,10 @@ public function getAllVersions(OperationContextInterface $context = null)
 
         $objectFactory = $this->getSession()->getObjectFactory();
         $result = array();
-        if (count($versions) !== null) {
+        if (count($versions)) {
             foreach ($versions as $objectData) {
                 $document = $objectFactory->convertObject($objectData, $context);
-                if (!(!$document instanceof DocumentInterface)) {
+                if (!($document instanceof DocumentInterface)) {
                     // this should never happen
```

Comment:
Could we use map to make these lines clearer in purpose?
When won't document instanceof DocumentInterface fulfill?